src/tests/src/libs/antares/study/series/timeseries-tests.cpp

```diff
@@ -0,0 +1,138 @@
+#define BOOST_TEST_MODULE "test time series"
+#define BOOST_TEST_DYN_LINK
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include <include/antares/series/series.h>
+
+using namespace Antares::Data;
+
+// =================
+// The fixture
+// =================
+struct Fixture
+{
+    Fixture(const Fixture& f) = delete;
+    Fixture(const Fixture&& f) = delete;
+    Fixture& operator=(const Fixture& f) = delete;
+    Fixture& operator=(const Fixture&& f) = delete;
+    Fixture() : ts(tsnum)
+    {
+        ts.resize(1, HOURS_PER_YEAR);
+        tsnum.resize(1, 1);
+    }
+    TimeSeries ts;
+    TimeSeries::TSNumbers tsnum;
+    std::string folder;
+
+    void fillColumn(unsigned int idx);
+    void fillColumnReverse(unsigned int idx);
+
+    void fillTsnum();
+
+};
+
+void Fixture::fillColumn(unsigned int idx)
+{
+    for (unsigned int i = 0; i < ts.timeSeries.height; i++)
+        ts.timeSeries[idx][i] = i;
+}
+
+void Fixture::fillColumnReverse(unsigned int idx)
+{
+    for (unsigned int i = 0; i < ts.timeSeries.height; i++)
+        ts.timeSeries[idx][i] = HOURS_PER_YEAR - i;
+}
+
+void Fixture::fillTsnum()
+{
+    tsnum.resize(1, ts.timeSeries.width);
+    for (unsigned int i = 0; i < ts.timeSeries.width; i++)
+        tsnum[0][i] = i;
+}
+
+// ==================
+// Tests section
+// ==================
+
+// Here, we need the "lightweight fixture"
+BOOST_AUTO_TEST_SUITE(timeseries_tests)
+
+BOOST_FIXTURE_TEST_CASE(getSeriesIndex, Fixture)
+{
+    tsnum.resize(1, 10);
+    for (unsigned int i = 0; i < 10; i++)
+        tsnum[0][i] = i;
+
+    //timeSeries.width == 1 so returns 0
+    BOOST_CHECK_EQUAL(ts.getSeriesIndex(5), 0);
```

Comment:
I'm surprised by this behaviour! Is it not a situation where we should throw ?

We have an index for series, but because we only have one timeseries, we don't take that index into account. it looks like an inconsistency in input data.

Let's not change it in this work, but I note it for later.